apps/api/src/domain/models/context/repositories/arango-note-query-repository.ts

```diff
@@ -0,0 +1,198 @@
+import {
+    IMultilingualTextItem,
+    LanguageCode,
+    PaginatedResponse,
+    ResourceCompositeIdentifier,
+} from '@coscrad/api-interfaces';
+import { isNullOrUndefined } from '@coscrad/validation-constraints';
+import { FetchManyQueryOptions } from '../../../../app/domain-modules/web-of-knowledge/interfaces/resource-query-repository.interface';
+import { InternalError } from '../../../../lib/errors/InternalError';
+import { Maybe } from '../../../../lib/types/maybe';
+import { isNotFound, NotFound } from '../../../../lib/types/not-found';
+import { ArangoConnectionProvider } from '../../../../persistence/database/arango-connection.provider';
+import { ArangoDatabase } from '../../../../persistence/database/arango-database';
+import { ArangoDatabaseForCollection } from '../../../../persistence/database/arango-database-for-collection';
+import convertResourceCompositeIdentifierToArangoDocumentHandle from '../../../../persistence/database/utilities/convertResourceCompositeIdentifierToArangoDocumentHandle';
+import { DTO } from '../../../../types/DTO';
+import { AggregateId } from '../../../types/AggregateId';
+import { MultilingualAudioItem } from '../../shared/multilingual-audio/multilingual-audio-item.entity';
+import { BaseArangoResourceViewQueryBuilder } from '../../term/repositories/base-arango-resource-query-builder';
+import { EventSourcedNoteViewModel } from '../event-sourced-note.view-model';
+import { INoteQueryRepository } from './note-query-repository.interface';
+
+const mapNoteViewDtoToArangoDocument = (dto: DTO<EventSourcedNoteViewModel>) => {
+    const { to: toMember, from: fromMember, self: selfMember } = dto.connectedResources;
+
+    const toAndFromCompositeIds: [ResourceCompositeIdentifier, ResourceCompositeIdentifier] =
+        isNullOrUndefined(selfMember)
+            ? [
+                  { type: toMember.type, id: toMember.id },
+                  { type: fromMember.type, id: fromMember.id },
+              ]
+            : [
+                  {
+                      type: selfMember.type,
+                      id: selfMember.id,
+                  },
+                  {
+                      type: selfMember.type,
+                      id: selfMember.id,
+                  },
+              ];
+
+    const [_to, _from] = toAndFromCompositeIds.map(
+        convertResourceCompositeIdentifierToArangoDocumentHandle
+    );
+
+    // TODO Use `toDto`
+    const document = {
+        ...dto,
+        _to,
+        _from,
+        _key: dto.id,
+    };
+
+    return document;
+};
+
+const mapArangoDocumentToNoteDto = (document) => {
+    const dto = {
```

Comment:
Do we really need to spread here? Or can we use a low level clone helper or else just `Object.assign` the `id` property directly to the document?

Reply:
I've done it that way for now.